From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 27 Jun 2024 18:15:00 +0000
Subject: Remove history-saving option for "Add an always-incognito mode"

Ultimately I think the option makes sense, but currently it is not
properly restricted to scenarios in which always-on incognito is
active. Even if it is prevented from being turned on except when
always-incognito is on, there is a window between that point and
relaunch in which it affects incognito behavior, and I strongly
believe it should *only* affect incognito for always-on incognito
to avoid breaking user expectations. That problem would need to be
resolved before bringing this back.

Change-Id: I446731c5b6c90519cdd71bbc4d0fbdd1db7e2ab0
---
 .../java/res/xml/incognito_preferences.xml       |  5 -----
 .../privacy/settings/IncognitoSettings.java      | 16 +---------------
 2 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/chrome/android/java/res/xml/incognito_preferences.xml b/chrome/android/java/res/xml/incognito_preferences.xml
--- a/chrome/android/java/res/xml/incognito_preferences.xml
+++ b/chrome/android/java/res/xml/incognito_preferences.xml
@@ -24,11 +24,6 @@
         android:title="@string/always_incognito_title"
         android:summary="@string/always_incognito_summary"
         android:defaultValue="false" />
-    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
-        android:key="incognito_history"
-        android:title="@string/incognito_history_enabled_title"
-        android:summary="@string/incognito_history_enabled_summary"
-        android:defaultValue="false" />
     <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
         android:key="incognito_save_site_setting"
         android:title="@string/incognito_save_site_setting_enabled_title"
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/IncognitoSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/IncognitoSettings.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/IncognitoSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/IncognitoSettings.java
@@ -55,12 +55,10 @@ public class IncognitoSettings
     private Snackbar mSnackbar;
 
     private static final String PREF_ALWAYS_INCOGNITO = "always_incognito";
-    private static final String PREF_INCOGNITO_TAB_HISTORY = "incognito_history";
     private static final String PREF_INCOGNITO_SAVE_SITE_SETTING = "incognito_save_site_setting";
 
     private final PrefService prefService = UserPrefs.get(ProfileManager.getLastUsedRegularProfile());
     private ChromeSwitchPreference alwaysIncognitoPref;
-    private ChromeSwitchPreference historyInIncognitoPref;
     private ChromeSwitchPreference saveSiteSettingsPref;
 
     @Override
@@ -72,8 +70,6 @@ public class IncognitoSettings
 
         setHasOptionsMenu(true);
 
-        historyInIncognitoPref =
-                (ChromeSwitchPreference) findPreference(PREF_INCOGNITO_TAB_HISTORY);
         saveSiteSettingsPref =
                 (ChromeSwitchPreference) findPreference(PREF_INCOGNITO_SAVE_SITE_SETTING);
         updatePreferences();
@@ -107,11 +103,6 @@ public class IncognitoSettings
                         /*actionData*/null)
                 .setDuration(/*durationMs*/70000);
 
-        historyInIncognitoPref.setChecked(!alwaysIncognito ? false :
-                prefService.getBoolean(Pref.INCOGNITO_TAB_HISTORY_ENABLED));
-        historyInIncognitoPref.setEnabled(alwaysIncognito);
-        historyInIncognitoPref.setOnPreferenceChangeListener(this);
-
         saveSiteSettingsPref.setChecked(!alwaysIncognito ? false :
                 prefService.getBoolean(Pref.INCOGNITO_SAVE_SITE_SETTING_ENABLED));
         saveSiteSettingsPref.setEnabled(alwaysIncognito);
@@ -121,21 +112,16 @@ public class IncognitoSettings
     @Override
     public boolean onPreferenceChange(Preference preference, Object newValue) {
         String key = preference.getKey();
+        prefService.setBoolean(Pref.INCOGNITO_TAB_HISTORY_ENABLED, false);
         if (PREF_ALWAYS_INCOGNITO.equals(key)) {
             if (((boolean) newValue) == false) {
-                prefService.setBoolean(Pref.INCOGNITO_TAB_HISTORY_ENABLED, false);
                 prefService.setBoolean(Pref.INCOGNITO_SAVE_SITE_SETTING_ENABLED, false);
-                historyInIncognitoPref.setChecked(false);
                 saveSiteSettingsPref.setChecked(false);
-                historyInIncognitoPref.setEnabled(false);
                 saveSiteSettingsPref.setEnabled(false);
             } else {
-                historyInIncognitoPref.setEnabled(true);
                 saveSiteSettingsPref.setEnabled(true);
             }
             AlwaysIncognitoLinkInterceptor.setAlwaysIncognito((boolean) newValue);
-        } else if (PREF_INCOGNITO_TAB_HISTORY.equals(key)) {
-            prefService.setBoolean(Pref.INCOGNITO_TAB_HISTORY_ENABLED, (boolean) newValue);
         } else if (PREF_INCOGNITO_SAVE_SITE_SETTING.equals(key)) {
             prefService.setBoolean(Pref.INCOGNITO_SAVE_SITE_SETTING_ENABLED, (boolean) newValue);
         }
-- 

